### PR TITLE
chore: send proposed release notes when feature-branch tests pass

### PR DESCRIPTION
**Pull request by Shesekino**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

send a summary of the feat/fix commits to a Slack channel so we can see what Semantic-Release will probably use to decorate the release once we merge to `staging`.
this helps us be more mindful of the release notes and their meaning to our users.

### More information

context
https://snyk.slack.com/archives/CDSMEJ29E/p158332105057200

### Screenshots

look at the bottom of the PR

### CODE REVIEW

**Comment by shaimendel** [comment]
I'm in favour of giving it a try :)

File: scripts/post-release-notes-preview.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+gitBranchPrefix = gitBranchParts[0]
+if not gitBranchPrefix in BRANCH_PREFIXES_TO_POST:
+    sys.exit(0)
+
+SLACK_WEBHOOK = os.environ['SLACK_WEBHOOK']
+USERNAME = os.environ.get('CIRCLE_USERNAME', 'USER_NOT_FOUND')
+
+# last version:
+lastVersion = subprocess.check_output(['git', 'describe', '--abbrev=0', '--tags', 'origin/staging'], text=True).strip()
```
Comment (ivanstanev) [inline scripts/post-release-notes-preview.py]: Hmm wondering if `--abbrev=0` is actually 3 separate arguments; seems to work either way!

File: scripts/post-release-notes-preview.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+gitBranchParts = GIT_BRANCH.split('/')
+if len(gitBranchParts) <= 1:
+    sys.exit(0)
+
+gitBranchPrefix = gitBranchParts[0]
+if not gitBranchPrefix in BRANCH_PREFIXES_TO_POST:
+    sys.exit(0)
+
+SLACK_WEBHOOK = os.environ['SLACK_WEBHOOK']
+USERNAME = os.environ.get('CIRCLE_USERNAME', 'USER_NOT_FOUND')
```
Comment (ivanstanev) [inline scripts/post-release-notes-preview.py]: Should not having a user ever happen?

File: scripts/post-release-notes-preview.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+# last version:
+lastVersion = subprocess.check_output(['git', 'describe', '--abbrev=0', '--tags', 'origin/staging'], text=True).strip()
+
+# commits since last version
+missingCommitsHeaders = subprocess.check_output(['git', 'log', '--no-decorate', lastVersion + '..HEAD', '--oneline'], text=True).strip()
```
Comment (ivanstanev) [inline scripts/post-release-notes-preview.py]: Wondering if the name is a bit misleading... perhaps commitsSinceLastVersion, just like the comment describes it?

File: scripts/post-release-notes-preview.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+features = []
+fixes = []
+others = []
+
+lines = missingCommitsHeaders.split('\n')
+for line in lines:
+    words = line.split(' ')
+    hash = words[0]
+    prefix = words[1]
+    rest = ' '.join(words[2:])
```
Comment (ivanstanev) [inline scripts/post-release-notes-preview.py]: What do we expect the rest to look like? Perhaps a comment with some sample content could help clarify here!

File: scripts/post-release-notes-preview.py (hunk 2 above)
Comment (Shesekino) [reply scripts/post-release-notes-preview.py]: when running locally :)
or... CircleCI freaking out?

File: scripts/post-release-notes-preview.py (hunk 4 above)
Comment (Shesekino) [reply scripts/post-release-notes-preview.py]: literally everything in the line output of ```git log --oneline --no-decorate``` after the hash and prefix:
so it's really just the commit message title :)

File: scripts/post-release-notes-preview.py (hunk 1 above)
Comment (Shesekino) [reply scripts/post-release-notes-preview.py]: :gandalf:

File: scripts/post-release-notes-preview.py (hunk 4 above)
Comment (Shesekino) [reply scripts/post-release-notes-preview.py]: there's no expectations around it - there's just collecting it and displaying it to us

File: scripts/post-release-notes-preview.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    commits = processCommits(commitsSinceLastVersion)
+
+    if not (commits['features'] or commits['fixes']):
+        sys.exit(0)
+
+    pullRequestUrls = os.environ.get('CIRCLE_PULL_REQUESTS', '')
+    if not pullRequestUrls:
+        sys.exit(0)
+
+    # TODO: support multiple PRs
```
Comment (ivanstanev) [inline scripts/post-release-notes-preview.py]: What does this mean? When can it occur?

Comment (Shesekino) [reply scripts/post-release-notes-preview.py]: - push commit to branch b1
- open PR from b1 to b2
- open PR from b1 to b3
🎉

**Pull request closed without merging**